kbase-extension/static/kbase/js/widgets/appWidgets2/validation.js

```diff
@@ -102,107 +181,38 @@ define([
-                        ).then((objectInfo) => {
-                            if (objectInfo) {
-                                const type = objectInfo.typeModule + '.' + objectInfo.typeName,
-                                    matchingType = options.types.some((typeId) => {
-                                        if (typeId === type) {
-                                            return true;
-                                        }
-                                        return false;
-                                    });
-                                if (!matchingType) {
-                                    messageId = 'obj-overwrite-diff-type';
-                                    errorMessage =
-                                        'an object already exists with this name and is not of the same type';
-                                    diagnosis = Constants.DIAGNOSIS.INVALID;
-                                } else {
-                                    messageId = 'obj-overwrite-warning';
-                                    shortMessage = 'an object already exists with this name';
-                                    diagnosis = Constants.DIAGNOSIS.SUSPECT;
-                                }
-                            }
-                        });
-                    }
+                ({ errorMessage, shortMessage, messageId, diagnosis, parsedValue } =
+                    validateWorkspaceObjectNameString(value, options));
+                if (errorMessage) {
+                    return;
                 }
-            }).then(() => {
-                return {
-                    isValid: errorMessage ? false : true,
-                    messageId,
-                    errorMessage,
-                    shortMessage,
-                    diagnosis,
-                    value,
-                    parsedValue,
-                };
-            });
-        }
-
-        function validateWorkspaceObjectRefSet(value, options) {
-            // TODO: validate each item.
-            let parsedValue,
-                messageId,
-                shortMessage,
-                errorMessage,
-                diagnosis = Constants.DIAGNOSIS.VALID;
-            return Promise.try(() => {
-                if (!(value instanceof Array)) {
-                    diagnosis = Constants.DIAGNOSIS.INVALID;
-                    errorMessage = 'value must be an array';
-                } else {
-                    parsedValue = value;
-                    if (parsedValue.length === 0) {
-                        if (options.required) {
-                            messageId = Constants.DIAGNOSIS.REQUIRED_MISSING;
-                            diagnosis = Constants.DIAGNOSIS.REQUIRED_MISSING;
-                            errorMessage = 'value is required';
-                        } else {
-                            diagnosis = Constants.DIAGNOSIS.OPTIONAL_EMPTY;
+                if (options.shouldNotExist) {
```

Comment:
Honestly, this logic could get re-sorted a bit, but it seemed better to mostly leave as-is for a major PR like this.